### PR TITLE
User UI

### PR DESCRIPTION
**Pull request by mazanec202**

### CODE REVIEW

**Review by paveltobias — COMMENTED** [review]
@mazanec202 Please reformat your changes by running `npm run format` in the `ui` directory. I strongly suggest fetching and rebasing on current `origin/main` (since there were some recent changes in Prettier config).

File: ui/src/app/components/users-table/users-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@Component({
+  selector: 'tc-users-table',
+  templateUrl: './users-table.component.html',
+  styleUrls: ['./users-table.component.scss'],
+})
+export class UsersTableComponent implements OnInit, OnDestroy {
+  readonly users$ = this.userService.orderedUsers$;
+  readonly displayedColumns: string[] = ['id', 'username', 'name', 'role'];
+
+  private readonly unsubscribe$ = new Subject<void>();
```
Comment (paveltobias) [inline ui/src/app/components/users-table/users-table.component.ts:15]: @mazanec202 

The usual purpose of this construct is unsubscribing from manual subscriptions (done via `Observable.subscribe()`) in component logic. E.g. in `court-detail.component.ts`:

```typescript
this.route.params.pipe(takeUntil(this.unsubscribe$)).subscribe(...);
```

The `takeUntil()` operator completes the observable chain (i.e. causes unsubscription) on the first value from `this.unsubscribe$`, which is done in `ngOnDestroy()`:

```typescript
ngOnDestroy(): void {
  this.unsubscribe$.next();
  this.unsubscribe$.complete();
}
```

When Angular destroys the view, it calls the `ngOnDestroy()` method. The `this.unsubscribe$.next()` call multicasts a sort-of "ping" (a `void`-type value) to all `takeUntil()` operators subscribed to it. That causes them to complete their respective observable chains and break the subscriptions. Without this, the subscriptions would "hang" after the component is destroyed.

In this component, no manual subscriptions making use of the `takeUntil()`-based unsubscription pattern are present. Therefore the `unsubscribe$` subject is redundant.

(Note: the `async` pipe, used to subscribe to observables in the template, "cleverly" unsubscribes on its own.)

File: ui/src/app/components/user-form/user-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  Role = 'Role',
+}
+
+@Component({
+  selector: 'tc-user-form',
+  templateUrl: './user-form.component.html',
+  styleUrls: ['./user-form.component.scss'],
+})
+export class UserFormComponent {
+  @Output() readonly cancelClick = new EventEmitter<void>();
```
Comment (paveltobias) [inline ui/src/app/components/user-form/user-form.component.ts:18]: This seems never to be used from outside.

File: ui/src/app/components/user-form/user-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      role: value[UserFormKey.Role],
+    };
+
+    this.userForm.markAsPristine();
+    this.userChange.emit(user);
+  }
+
+  cancel(): void {
+    this.cancelClick.emit();
+  }
```
Comment (paveltobias) [inline ui/src/app/components/user-form/user-form.component.ts:62]: This seems never to be used in the template.

File: ui/src/app/views/main/users-overview/users-overview.component.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'tc-users-overview',
+  templateUrl: './users-overview.component.html',
+  styleUrls: ['./users-overview.component.scss'],
+})
+export class UsersOverviewComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (paveltobias) [inline ui/src/app/views/main/users-overview/users-overview.component.ts:11]: Dead code.

**Review by paveltobias — CHANGES_REQUESTED** [review]

File: ui/src/app/components/users-table/users-table.component.ts (hunk 1 above)
Comment (mazanec202) [reply ui/src/app/components/users-table/users-table.component.ts:15]: I wasn't aware of this, thanks!

**Review by paveltobias — APPROVED** [review]

**Pull request merged**